sys/kern/kern_exec.c

```diff
@@ -1922,7 +1922,8 @@ exec_copyout_strings(struct image_params *imgp, uintcap_t *stack_base)
 #else
 		imgp->auxv = vectp;
```

Comment:
I feel like this should be a cheri_kern_setbounds() instead of the ifdef. I also find myself wanting `AT_COUNT * sizeof(Elf_Auxinfo)` to be imgp->auxvlen or some such so we can pad it to allow setbounds_exact (not that we're anywhere near that today).

Reply:
`cheri_kern_*` are nops in a hybrid kernel.  It is true that maybe in the kernel we could have a set of `cheri_user_*` wrappers that are nops in non-CHERI kernels and `cheri_*` in both hybrid and purecap kernels.  That might let us remove several #ifdef's in various places.

Reply:
Adding an auxvlen might indeed be nice but would be a future change I think.  Maybe something to put on the TODO list.